benchmarks/distributed/lambda_runtimes/cpp_runtime/main.cpp

```diff
@@ -0,0 +1,217 @@
+struct Timestamps {
+    uint64_t MAIN_TS;
+    uint64_t READY_TS;
+    uint64_t PYTHON_READY_TS;
+    uint64_t FUNCTION_TS;
+    bool reused;
+
+    void setString(double func_res) {
+        std::stringstream ss;
+        ss << "{"
+           << "\"func_res\":" << func_res
+           << ", \"reused\":" << reused
+           << ", \"body\": {"
+             << "\"MAIN_TS\":" << MAIN_TS
+             << ", \"READY_TS\":" << READY_TS
+             << ", \"PYTHON_READY_TS\":" << PYTHON_READY_TS
+             << ", \"FUNCTION_TS\":" << FUNCTION_TS
+           << "}}";
+        reused = true; // next time, it will be true
+
+        res = ss.str();
+    }
+
+    Timestamps() : MAIN_TS(0), READY_TS(0), PYTHON_READY_TS(0), FUNCTION_TS(0), reused(false) {}
+};
+
+Timestamps timings;
+
+static std::string decodeFunction(const std::string &func_str, int &r) {
+    // do some stupid stuff to get around not having protobuf
+    auto b64Module = PyImport_ImportModule("base64");
+    if(!b64Module) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't open module base64";
+    }
+    auto b64decode = PyObject_GetAttrString(b64Module, "b64decode");
+    if (!b64decode) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't get function base64.b64decode";
+    }
+    auto py_pickled_func_bytes = PyObject_CallFunctionObjArgs(b64decode, PyUnicode_FromString(func_str.c_str()), nullptr);
+    if(!py_pickled_func_bytes) {
+        PyErr_Print();
+        r = -1;
+        return "base64.b64decode failed!";
+    }
+//    std::cout << "pickled func addr: "<< (void*)py_pickled_func_bytes << std::endl;
+    char *mem = PyBytes_AsString(py_pickled_func_bytes);
+    if(!mem) {
+        PyErr_Print();
+        r = -1;
+        return "decoded function wasn't bytes?";
+    }
+    size_t mem_size = PyBytes_Size(py_pickled_func_bytes);
+    std::string res;
+    res.reserve(mem_size);
+    res.assign(mem, mem_size);
+    return res;
+}
+
+const char* handler_worker(const char *payload, int &r) {
+    using namespace Aws::Utils::Json;
+    timings.READY_TS = timestamp();
+    r = 0; // no error
+
+    // extract + set python home
+    auto task_root = std::getenv("LAMBDA_TASK_ROOT");
+    // convert to wchar
+    std::vector<wchar_t> vec;
+    auto len = strlen(task_root);
+    vec.resize(len + 1);
+    mbstowcs(&vec[0], task_root, len);
+    Py_SetPythonHome(&vec[0]);
+
+    // initialize interpreter
+    // Py_Initialize();
+    if (!Py_IsInitialized()) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't initialize interpreter!";
+    }
+    timings.PYTHON_READY_TS = timestamp();
+
+    // grab the payload
+    JsonValue json(payload);
+    auto v = json.View();
+    auto func_str = v.GetString("function");
+    auto pickled_func_str = decodeFunction(func_str, r);
+    if (r == -1) {
+        res = pickled_func_str;
+        return res.c_str();
+    }
+
+    /*auto py_func_str = PyUnicode_FromString(func_str.c_str());
```

Comment:
remove commented code or explain what it does

Reply:
added comment